sdk/core/azure-core/src/http/http.cpp

```diff
@@ -0,0 +1,157 @@
+      0, /* 26 -  substitute */
+      0, /* 27 -  escape */
+      0, /* 28 -  file separator */
+      0, /* 29 -  group separator */
+      0, /* 30 -  record separator */
+      0, /* 31 -  unit separator */
+      ' ', /* 32 -  space */
+      '!', /* 33 -  ! */
+      0, /* 34 -  " */
+      '#', /* 35 -  # */
+      '$', /* 36 -  $ */
+      '%', /* 37 -  % */
+      '&', /* 38 -  & */
+      '\'', /* 39 - ' */
+      0, /* 40 -  ( */
+      0, /* 41 -  ) */
+      '*', /* 42 -  * */
+      '+', /* 43 -  + */
+      0, /* 44 -  , */
+      '-', /* 45 -  - */
+      '.', /* 46 -  . */
+      0, /* 47 -  / */
+      '0', /* 48 -  0 */
+      '1', /* 49 -  1 */
+      '2', /* 50 -  2 */
+      '3', /* 51 -  3 */
+      '4', /* 52 -  4 */
+      '5', /* 53 -  5 */
+      '6', /* 54 -  6 */
+      '7', /* 55 -  7 */
+      '8', /* 56 -  8 */
+      '9', /* 57 -  9 */
+      0, /* 58 -  : */
+      0, /* 59 -  ; */
+      0, /* 60 -  < */
+      0, /* 61 -  = */
+      0, /* 62 -  > */
+      0, /* 63 -  ? */
+      0, /* 64 -  @ */
+      'a', /* 65 -  A */
+      'b', /* 66 -  B */
+      'c', /* 67 -  C */
+      'd', /* 68 -  D */
+      'e', /* 69 -  E */
+      'f', /* 70 -  F */
+      'g', /* 71 -  G */
+      'h', /* 72 -  H */
+      'i', /* 73 -  I */
+      'j', /* 74 -  J */
+      'k', /* 75 -  K */
+      'l', /* 76 -  L */
+      'm', /* 77 -  M */
+      'n', /* 78 -  N */
+      'o', /* 79 -  O */
+      'p', /* 80 -  P */
+      'q', /* 81 -  Q */
+      'r', /* 82 -  R */
+      's', /* 83 -  S */
+      't', /* 84 -  T */
+      'u', /* 85 -  U */
+      'v', /* 86 -  V */
+      'w', /* 87 -  W */
+      'x', /* 88 -  X */
+      'y', /* 89 -  Y */
+      'z', /* 90 -  Z */
+      0, /* 91 -  [ */
+      0, /* 92 -  comment */
+      0, /* 93 -  ] */
+      '^', /* 94 -  ^ */
+      '_', /* 95 -  _ */
+      '`', /* 96 -  ` */
+      'a', /* 97 -  a */
+      'b', /* 98 -  b */
+      'c', /* 99 -  c */
+      'd', /* 100 -  d */
+      'e', /* 101 -  e */
+      'f', /* 102 -  f */
+      'g', /* 103 -  g */
+      'h', /* 104 -  h */
+      'i', /* 105 -  i */
+      'j', /* 106 -  j */
+      'k', /* 107 -  k */
+      'l', /* 108 -  l */
+      'm', /* 109 -  m */
+      'n', /* 110 -  n */
+      'o', /* 111 -  o */
+      'p', /* 112 -  p */
+      'q', /* 113 -  q */
+      'r', /* 114 -  r */
+      's', /* 115 -  s */
+      't', /* 116 -  t */
+      'u', /* 117 -  u */
+      'v', /* 118 -  v */
+      'w', /* 119 -  w */
+      'x', /* 120 -  x */
+      'y', /* 121 -  y */
+      'z', /* 122 -  z */
+      0, /* 123 -  { */
+      '|', /* 124 -  | */
+      0, /* 125 -  } */
+      '~', /* 126 -  ~ */
+      0 /* 127 -  DEL */
+      // ...128-255 is all zeros (not valid) characters}
+  };
+
+  // Check all chars in name are valid
+  for (size_t index = 0; index < headerName.size(); index++)
+  {
+    if (validChars[static_cast<int>(headerName[index])] == 0)
+    {
+      return index + 1;
```

Comment:
Why are we adding 1 here? Shouldn't it just be return `index`?

Reply:
We return 0 on success. This makes index 0 to be reserved
So the index we returns starts at 1.

Reply:
What is the purpose of returning an incorrect index value, then? Why not just return -1, or a bool for success/fail?